codemetrics.el

```diff
@@ -390,6 +408,11 @@ For argument NODE, see function `codemetrics-analyze' for more information."
   "Handle recursion for most languages uses `identifier' as the keyword."
   (codemetrics-rules--recursion-using-node-name node "identifier"))
 
+;; bash uses its own name for its function invocation names: command_name
+(defun codemetrics-rules--bash-recursion (node &rest _)
+  "Handle recursion for bash."
+  (codemetrics-rules--recursion-using-node-name node "command_name"))
```

Comment:
IIRC, we can use `lambda` in the rule function. So we don't have to declare these one-line functions.

Reply:
Good idea! 😄 Refactored in my latest commit.

Reply:
Can you do this the same to all one-line functions:

- `codemetrics-rules--kotlin-function-declaration`
- `codemetrics-rules--kotlin-recursion`
- `codemetrics-rules--kotlin-elvis-operator`

The general rule is to use `lambda` when the rule only works in one place (a very specific condition and no duplicate code is produced). But if a function like `codemetrics-rules--java-outer-loop` is used in multiple parsers, then it can have its own function.

Reply:
Done in latest commit 🙂 The reason I did not do it right away was because I tried to keep the commits within the intent of the PR (i.e, bash stuff).

Reply:
Yeah, you are right! It's okay in this scenario since you are the author of both PRs. :)